src/librustc_mir/diagnostics.rs

```diff
@@ -1978,8 +1978,8 @@ could cause the match to be non-exhaustive:
 let mut x = Some(0);
 match x {
     None => (),
-    Some(v) if { x = None; false } => (),
-    Some(_) => (), // No longer matches
```

Comment:
I don't understand what motivated this change... isn't this example rejected regardless of whether one includes a binding of `v` or leaves it out?

Reply:
Or I guess that is in fact the whole point: That your PR changes things so that the previous version is now accepted.

Reply:
The PR changes the error of the previous example to a normal borrow conflicts error. This is the smallest change that makes this fall with E0510.

Reply:
Ah okay, sounds great.